src/main/java/org/testng/internal/ConfigurationGroupMethods.java

```diff
@@ -37,18 +37,14 @@
   private volatile Map<String, List<ITestNGMethod>> m_afterGroupsMap = null;
 
   public ConfigurationGroupMethods(
-      ITestNGMethod[] allMethods,
+      IContainer<ITestNGMethod> container,
       Map<String, List<ITestNGMethod>> beforeGroupsMethods,
       Map<String, List<ITestNGMethod>> afterGroupsMethods) {
-    m_allMethods = allMethods;
+    m_allMethods = container.getItems();

```

Comment:
For now, ok but the cache modification won't be triggered here.
`m_allMethods` should be replaced by `container`

Reply:
The cache is never meant to be modified so this is fine.